services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/FileBasedCredentialsService.java

```diff
@@ -322,6 +324,17 @@ private JsonObject getSingleCredentials(final String tenantId, final String auth
                     final JsonObject authIdCredential = (JsonObject) authIdCredentialEntry;
                     // return the first matching type entry for this authId
                     if (type.equals(authIdCredential.getString(CredentialsConstants.FIELD_TYPE))) {
+                        if (clientContext != null) {
```

Comment:
FMPOV we do not need to process the client context information at all In our example Device Registry. We mainly introduced it so that other implementations could use the information for whatever purpose they see fit.

Reply:
I think it's valuable to have it tested at least in the most simple form, so that things don't get broken over time. It doesn't compromise anything else, so I'm inclined to leave it in.

Reply:
My point is that if we do not specify any expected processing of the extra information in the spec (which we don't/shouldn't), then there actually is nothing to test, right? Or would you like to specify the behavior that you are implementing here as the *expected behavior* of at least our example Device Registry (if not of the operation in general)?

Reply:
My concern is if it's just ignored and not tested anywhere, someone will be inclined to just deleted after a while. Also, this gives an example of "this feature" to the people that wants to implement/adapt device registries to Hono API. It doesn't change anything for the folks that don't care about it, so I don't see a problem there.

Reply:
Ok, then please don't forget to describe this behavior accordingly in the Device Registry User Guide.

Reply:
This change is not propagated to the default device registry implementation. It's just in the core API at the moment. It might get useful to add it as part of the integration work at some point, but I'm inclined to leave it as is for now. WDYT?

Reply:
I don't understand your point. What do you mean by
> This change is not propagated to the default device registry implementation. It's just in the core API at the moment.

The code you are working on is part of the Device Registry component we ship with Hono.

Reply:
I meant that the HTTP API of Device Registry don't handle these extra parameters and there's no integration test for this functionality (over HTTP). I can add those for completeness.

Reply:
Now I am completely confused. What does the HTTP API have to do with this? Do you mean that we cannot register this kind of additional information using the HTTP API?